frontend/pages/packs/EditPackPage/EditPackPage.tsx

```diff
@@ -0,0 +1,371 @@
+    {
+      select: (data: IStoredLabelsResponse) => data.labels,
+    }
+  );
+
+  const packLabels = storedPack
+    ? filter(labels, (label) => {
+        return includes(storedPack.label_ids, label.id);
+      })
+    : [];
+
+  const {
+    isLoading: isHostsLoading,
+    data: hosts,
+    error: hostsError,
+  } = useQuery<IStoredHostsResponse, Error, IHost[]>(
+    ["all hosts"], // use single string or array of strings can be named anything
+    () => hostsAPI.loadAll(undefined),
+    {
+      select: (data: IStoredHostsResponse) => data.hosts,
+    }
+  );
+  const packHosts = storedPack
+    ? filter(hosts, (host) => {
+        return includes(storedPack.host_ids, host.id);
+      })
+    : [];
+
+  const {
+    isLoading: isTeamsLoading,
+    data: teams,
+    error: teamsError,
+  } = useQuery<IStoredTeamsResponse, Error, ITeam[]>(
+    ["pack labels"],
+    () => teamsAPI.loadAll(),
+    {
+      select: (data: IStoredTeamsResponse) => data.teams,
+    }
+  );
+
+  const packTeams = storedPack
+    ? filter(teams, (team) => {
+        return includes(storedPack.team_ids, team.id);
+      })
+    : [];
+
+  const packTargets = [...packHosts, ...packLabels, ...packTeams];
+
+  const onCancelEditPack = () => {
+    return dispatch(push(PATHS.MANAGE_PACKS));
+  };
+
+  const onFetchTargets = (
+    query: IQuery,
+    targetsResponse: ITargetsAPIResponse
+  ) => {
+    const { targets_count } = targetsResponse;
+
+    setTargetsCount(targets_count);
+
+    return false;
+  };
+
+  const togglePackQueryEditorModal = useCallback(() => {
+    setSelectedPackQuery(undefined); // create modal renders
+    setShowPackQueryEditorModal(!showPackQueryEditorModal);
+    console.log("togglePackQueryEditorModal clicked!");
+  }, [showPackQueryEditorModal, setShowPackQueryEditorModal]);
+
+  const toggleEditPackQueryModal = useCallback(() => {
+    setShowEditPackQueryModal(!showEditPackQueryModal);
+  }, [showEditPackQueryModal, setShowEditPackQueryModal]);
+
+  const toggleRemovePackQueryModal = useCallback(() => {
+    setShowRemovePackQueryModal(!showRemovePackQueryModal);
+  }, [showRemovePackQueryModal, setShowRemovePackQueryModal]);
+
+  const onEditPackQueryClick = (selectedQuery: any): void => {
+    togglePackQueryEditorModal();
+    setSelectedPackQuery(selectedQuery); // edit modal renders
+  };
+
+  const onRemovePackQueriesClick = (selectedTableQueryIds: any): void => {
+    toggleRemovePackQueryModal();
+    setSelectedPackQueryIds(selectedTableQueryIds);
+  };
+
+  const handlePackFormSubmit = (formData: any) => {
+    const updatedPack = deepDifference(formData, storedPack);
+    packsAPI
+      .update(packId, updatedPack)
+      .then(() => {
+        dispatch(renderFlash("success", `Successfully updated this pack.`));
+      })
+      .catch(() => {
+        dispatch(
+          renderFlash("error", `Could not update pack. Please try again.`)
+        );
+      });
+  };
+
+  const onPackQueryEditorSubmit = useCallback(
```

Comment:
I don't think useCallback is necessary. The function is run in a controlled manner.